drivers/led/is31fl3194/is31fl3194.c

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) 2022 Benjamin Björnsson <benjamin.bjornsson@gmail.com>.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT issi_is31fl3194
+
+/**
+ * @file
+ * @brief LED driver for the IS31FL3194 I2C LED driver
+ */
+
+#include <zephyr/zephyr.h>
+#include <zephyr/drivers/led.h>
+#include <zephyr/drivers/i2c.h>
+#include <string.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(is31fl3194, CONFIG_LED_LOG_LEVEL);
+
+#include "is31fl3194_reg.h"
+
+#define IS31FL3194_INDEX_MAX 2
+
+struct is31fl3194_config {
+	struct is31fl3194_ctx ctx;
+	const struct i2c_dt_spec i2c;
+	uint8_t num_leds;
+	const struct led_info *leds_info;
+	const uint8_t *curr_map_max;
+};
+
+struct is31fl3194_data {
+	struct is31fl3194_regs regs;
+};
+
+int is31fl3194_i2c_read(const struct i2c_dt_spec *i2c_spec, uint8_t reg_addr, uint8_t *value,
+			uint8_t len)
+{
+	return i2c_burst_read_dt(i2c_spec, reg_addr, value, len);
+}
+
+int is31fl3194_i2c_write(const struct i2c_dt_spec *i2c_spec, uint8_t reg_addr, uint8_t *value,
+			 uint8_t len)
+{
+	return i2c_burst_write_dt(i2c_spec, reg_addr, value, len);
+}
+
+int is31fl3194_on(const struct device *dev, uint32_t led)
+{
+	const struct is31fl3194_config *config = dev->config;
+	struct is31fl3194_ctx *ctx = (struct is31fl3194_ctx *)&config->ctx;
+	int res;
+
+	res = is31fl3194_out_en_set(ctx, led, 1);
+	if (res) {
+		return res;
+	}
+
+	return is31fl3194_color_update(ctx);
+}
+
+int is31fl3194_off(const struct device *dev, uint32_t led)
+{
+	const struct is31fl3194_config *config = dev->config;
+	struct is31fl3194_ctx *ctx = (struct is31fl3194_ctx *)&config->ctx;
+	int res;
+
+	res = is31fl3194_out_en_set(ctx, led, 0);
+	if (res) {
+		return res;
+	}
+
+	return is31fl3194_color_update(ctx);
+}
+
+static const struct led_info *is31fl3194_led_to_info(const struct is31fl3194_config *config,
+						     uint32_t led)
+{
+	for (int i = 0; i < config->num_leds; i++) {
+		if (config->leds_info[i].index == led) {
+			return &config->leds_info[i];
+		}
+	}
+
+	return NULL;
+}
+
+static int is31fl3194_get_info(const struct device *dev, uint32_t led, const struct led_info **info)
+{
+	const struct is31fl3194_config *config = dev->config;
+	const struct led_info *leds_info;
+
+	leds_info = is31fl3194_led_to_info(config, led);
+	if (!leds_info) {
+		*info = NULL;
+		return -EINVAL;
+	}
+
+	*info = leds_info;
+
+	return 0;
+}
+
+int is31fl3194_set_brightness(const struct device *dev, uint32_t led, uint8_t value)
```

Comment:
In led.h it is written that the brightness value is a percent. So I think you have to convert the brightness value from the [0-100] range to [0-255].

Note that the API should be changed because there is no reason to loose granularity on brightness.

Reply:
Agreed, will updated this.

I agree, makes more sense to use full range of `uint8_t`.